pkg/policyprovider/configpolicy/configpolicy_test.go

```diff
@@ -80,3 +86,168 @@ func TestPolicyEnforcer_ContinueVerifyOnFailure(t *testing.T) {
 		t.Fatalf("For artifact types without a policy the default policy should be followed")
 	}
 }
+
+func TestPolicyEnforcer_OverallVerifyResult(t *testing.T) {
+
+	testcases := []struct {
+		configPolicyConfig map[string]interface{}
+		verifierReports    []interface{}
+		output             bool
+	}{
+		{
+			// no artifact policies or verifier reports
+			configPolicyConfig: map[string]interface{}{
+				"name":                         "configPolicy",
+				"artifactVerificationPolicies": map[string]types.ArtifactTypeVerifyPolicy{},
+			},
+			verifierReports: []interface{}{},
+			output:          true,
+		},
+		{
+			// no artifact policies
+			configPolicyConfig: map[string]interface{}{
+				"name":                         "configPolicy",
+				"artifactVerificationPolicies": map[string]types.ArtifactTypeVerifyPolicy{},
+			},
+			verifierReports: []interface{}{
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    true,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+			},
+			output: true,
+		},
+		{
+			// any notary artifact policy and only 1 notary report is true
+			configPolicyConfig: map[string]interface{}{
+				"name": "configPolicy",
+				"artifactVerificationPolicies": map[string]types.ArtifactTypeVerifyPolicy{
+					"application/vnd.cncf.notary.v2": "any",
+				},
+			},
+			verifierReports: []interface{}{
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    true,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    false,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+			},
+			output: true,
+		},
+		{
+			// all notary artifact policy but only 1 notary report is true
+			configPolicyConfig: map[string]interface{}{
+				"name": "configPolicy",
+				"artifactVerificationPolicies": map[string]types.ArtifactTypeVerifyPolicy{
+					"application/vnd.cncf.notary.v2": "all",
+				},
+			},
+			verifierReports: []interface{}{
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    true,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+				vr.VerifierResult{
+					Subject:      "",
+					IsSuccess:    false,
+					ArtifactType: "application/vnd.cncf.notary.v2",
+				},
+			},
+			output: false,
+		},
+		{
+			// all notary artifact policy and both notary reports are true
+			configPolicyConfig: map[string]interface{}{
+				"name": "configPolicy",
+				"artifactVerificationPolicies": map[string]types.ArtifactTypeVerifyPolicy{
+					"application/vnd.cncf.notary.v2": "all",
+				},
+			},
+			verifierReports: []interface{}{
```

Comment:
is there a test case for notary required, but no artifact available?

Reply:
I missed this case. I'll add it.